components/brave_wallet/browser/ethereum_provider_impl.cc

```diff
@@ -108,22 +108,14 @@ void RejectMismatchError(base::Value id,
                           false);
 }
 
-bool IsTypedDataStructure(const std::string& normalized_json_request) {
-  std::string address;
-  std::string message;
-  base::Value::Dict domain;
-  std::vector<uint8_t> domain_hash_out;
-  std::vector<uint8_t> primary_hash_out;
-
-  mojom::EthSignTypedDataMetaPtr meta;
-
-  if (ParseEthSignTypedDataParams(normalized_json_request, &address, &message,
-                                  &domain, EthSignTypedDataHelper::Version::kV4,
-                                  &domain_hash_out, &primary_hash_out, &meta) ||
-      ParseEthSignTypedDataParams(normalized_json_request, &address, &message,
-                                  &domain, EthSignTypedDataHelper::Version::kV3,
-                                  &domain_hash_out, &primary_hash_out, &meta)) {
-    return true;
+bool IsTypedDataStructure(const base::Value::List& params_list) {
+  if (ParseEthSignTypedDataAddress(params_list)) {
+    if (ParseEthSignTypedDataParams(params_list,
```

Comment:
Would it be better to include address in `EthSignTypedData` and just parse it in `ParseEthSignTypedDataParams`?
Will need to revise `SignTypedMessage`, but seems easier to read to me and we don't need to do address null checks outside of the parser function.

Reply:
Added https://github.com/brave/brave-core/pull/26165/commits/77caacd564fe1cc0b8620d9d090be0da54030f00